src/iptvsimple/utilities/FileUtils.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ *      Copyright (C) 2005-2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, write to
+ *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+
+#include "FileUtils.h"
+
+#include "../Settings.h"
+#include "../../client.h"
+#include "zlib.h"
+
+using namespace iptvsimple;
+using namespace iptvsimple::utilities;
+
+std::string FileUtils::PathCombine(const std::string& path, const std::string& fileName)
+{
+  std::string result = path;
+  if (result.at(result.size() - 1) == '\\' ||
+      result.at(result.size() - 1) == '/')
+  {
+    result.append(fileName);
+  }
+  else
+  {
+    result.append("/");
+    result.append(fileName);
+  }
+
+  return result;
+}
+
+std::string FileUtils::GetClientFilePath(const std::string& fileName)
+{
+  return PathCombine(Settings::GetInstance().GetClientPath(), fileName);
+}
+
+std::string FileUtils::GetUserFilePath(const std::string& fileName)
+{
+  return PathCombine(Settings::GetInstance().GetUserPath(), fileName);
+}
+
+int FileUtils::GetFileContents(const std::string& url, std::string& content)
+{
+  content.clear();
+  void* fileHandle = XBMC->OpenFile(url.c_str(), 0);
+  if (fileHandle)
+  {
+    char buffer[1024];
+    while (int bytesRead = XBMC->ReadFile(fileHandle, buffer, 1024))
+      content.append(buffer, bytesRead);
+    XBMC->CloseFile(fileHandle);
+  }
+
+  return content.length();
+}
+
+/*
+ * This method uses zlib to decompress a gzipped file in memory.
+ * Author: Andrew Lim Chong Liang
+ * http://windrealm.org
+ */
+
+bool FileUtils::GzipInflate(const std::string& compressedBytes, std::string& uncompressedBytes)
+{
+  if (compressedBytes.size() == 0)
+  {
+    uncompressedBytes = compressedBytes;
+    return true;
+  }
+
+  uncompressedBytes.clear();
+
+  unsigned full_length = compressedBytes.size();
+  unsigned half_length = compressedBytes.size() / 2;
```

Comment:
Could be `const unsigned` to make clear that this method will not modify this value.